develop/app/Filters/UserFilter.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Filters;
+
+use CodeIgniter\Filters\FilterInterface;
+use CodeIgniter\HTTP\RequestInterface;
+use CodeIgniter\HTTP\ResponseInterface;
+use App\Services\User;
+
+class UserFilter implements FilterInterface
+{
+    private $response;
+
+    public function __construct()
+    {
+        $this->response = \CodeIgniter\Config\Services::response();
+    }
+
+    /**
+     * Do whatever processing this filter needs to do.
+     * By default it should not return anything during
+     * normal execution. However, when an abnormal state
+     * is found, it should return an instance of
+     * CodeIgniter\HTTP\Response. If it does, script
+     * execution will end and that Response will be
+     * sent back to the client, allowing for error pages,
+     * redirects, etc.
+     *
+     * @param RequestInterface $request
+     * @param array|null       $arguments
+     *
+     * @return void
+     */
+    public function before(RequestInterface $request, $arguments = null)
+    {
+        $request = \Config\Services::request();
+
+        $user_key = $request->getHeaderLine("X-User-Key");
+
+        $failBody = [
+            "status" => 401,
+            "error"  => 401,
+            "message" => [
+                "error" => "The user not valid!"
+            ]
+        ];
+        if ($user_key == "") {
```

Comment:
Is it more safety by add the `is_null()` condition? 
And we need to assume we have these users, like `user_key` 1, 2, 3 et al.
We implement `user` service in the simplest way, but you should need to authenticate users.